[CIVIS] Revamp setup instructions for Gitlab (#20897)
DataDog/documentation: content/en/continuous_integration/pipelines/gitlab.md

[2]: https://docs.gitlab.com/ee/administration/operations/rails_console.html#using-the-rails-runner
[3]: https://docs.gitlab.com/ee/user/admin_area/settings/project_integration_management.html#use-custom-settings-for-a-group-or-project-integration
{{% /tab %}}

{{% tab "GitLab &lt; 13.7" %}}

> For versions older than when the native Datadog integration, you can use [webhooks][1] to send pipeline data to Datadog.

Reviewer: ```suggestion
For older versions of GitLab, you can use [webhooks][1] to send pipeline data to Datadog.
```

> <div class="alert alert-info"><strong>Note</strong>: Support directly with webhooks is not actively under development. Unexpected issues could happen. Updating the GitLab version is recommended instead.</div>

Reviewer: ```suggestion
<div class="alert alert-info"><strong>Note</strong>: Direct support with webhooks is not under development. Unexpected issues could happen. Datadog recommends that you update GitLab instead.</div>
```

The following GitLab versions support collecting job logs:

* GitLab.com (SaaS)
* GitLab >= 15.3 (self-hosted) only if you are using [object storage to store job logs][7]
* GitLab >= 14.8 (self-hosted) by enabling the `datadog_integration_logs_collection` feature flag

<div class="alert alert-info"><strong>Note</strong>: Logs are billed separately from CI Visibility.</div>

Job logs are collected in [Log Management][9] and are automatically correlated with the GitLab pipeline in CI Visibility. Log files larger than one GiB are truncated.

For more information about processing job logs collected from the GitLab integration, see the [Processors documentation][17].

To enable collection of job logs:

{{< tabs >}}
{{% tab "GitLab.com" %}}

> 1. Click **Enable job logs collection** checkbox in the GitLab integration **Settings > Integrations > Datadog**.

Reviewer: ```suggestion
1. Click the **Enable job logs collection** checkbox in the GitLab integration **Settings > Integrations > Datadog**.
```

2. Click **Save changes**.
{{% /tab %}}

{{% tab "GitLab &gt;&equals; 15.3" %}}

> <div class="alert alert-info"><strong>Note</strong>: Datadog downloads log files directly from your GitLab logs [object storage][1] with temporary pre-signed URLs.

Reviewer: ```suggestion
<div class="alert alert-info"><strong>Note</strong>: Datadog downloads log files directly from your GitLab logs <a href="https://docs.gitlab.com/ee/administration/job_artifacts.html#using-object-storage">object storage</a> with temporary pre-signed URLs.
```

> The storage must not have network restrictions such as IP range whitelist.</div>

Reviewer: ```suggestion
The storage must not have network restrictions, such as an IP range allowlist.</div>
```

> 1. Click **Enable job logs collection** checkbox in the GitLab integration **Settings > Integrations > Datadog**.

Reviewer: ```suggestion
1. Click **Enable job logs collection** checkbox in the GitLab integration under **Settings > Integrations > Datadog**.
```

2. Click **Save changes**.

> [1]: https://docs.gitlab.com/ee/administration/job_artifacts.html#using-object-storage

Reviewer: ```suggestion
```

{{% /tab %}}

{{% tab "GitLab &gt;&equals; 14.8" %}}

> 1. Enable the `datadog_integration_logs_collection` [feature flag][1] in your GitLab. This allows you to see the **Enable job logs collection** checkbox in the GitLab integration **Settings > Integrations > Datadog**.

Reviewer: ```suggestion
1. Enable the `datadog_integration_logs_collection` [feature flag][1] in your GitLab. This allows you to see the **Enable job logs collection** checkbox in the GitLab integration under **Settings > Integrations > Datadog**.
```
